sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileAsyncApiTests.java

```diff
@@ -0,0 +1,435 @@
+
+        StepVerifier.create(fc.create())
+            .assertNext(r -> assertNotEquals(null, r))
+            .verifyComplete();
+    }
+
+    @Test
+    public void createDefaults() {
+        fc = dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+
+        StepVerifier.create(fc.createWithResponse(null, null, null, null, null))
+            .assertNext(r -> assertEquals(201, r.getStatusCode()))
+            .verifyComplete();
+    }
+
+    @Test
+    public void createError() {
+        fc = dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+
+        StepVerifier.create(fc.createWithResponse(null, null, null, null, new DataLakeRequestConditions().setIfMatch("garbage")))
+            .verifyError(DataLakeStorageException.class);
+    }
+
+    @Test
+    public void createOverwrite() {
+        fc = dataLakeFileSystemAsyncClient.createFile(generatePathName()).block();
+
+        StepVerifier.create(fc.create(false))
+            .verifyError(DataLakeStorageException.class);
+    }
+
+    @Test
+    public void exists() {
+        fc = dataLakeFileSystemAsyncClient.createFile(generatePathName()).block();
+
+        StepVerifier.create(fc.exists())
+            .expectNext(true)
+            .verifyComplete();
+    }
+
+    @Test
+    public void doesNotExist() {
+        fc = dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+
+        StepVerifier.create(fc.exists())
+            .expectNext(false)
+            .verifyComplete();
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {"null,null,null,null,null", "control,disposition,encoding,language,type"})
+    public void createHeaders(String cacheControl, String contentDisposition, String contentEncoding,
+                              String contentLanguage, String contentType) {
+        // Create does not set md5
+        PathHttpHeaders headers = new PathHttpHeaders().setCacheControl(cacheControl)
+            .setContentDisposition(contentDisposition)
+            .setContentEncoding(contentEncoding)
+            .setContentLanguage(contentLanguage)
+            .setContentType(contentType);
+
+        fc = dataLakeFileSystemAsyncClient.getFileAsyncClient(generatePathName());
+
+        contentType = (contentType == null) ? "application/octet-stream" : contentType;
+        fc.createWithResponse(null, null, headers, null, null).block();
+
+        String finalContentType = contentType;
+        StepVerifier.create(fc.getPropertiesWithResponse(null))
+            .assertNext(r -> {
+                assertEquals(cacheControl, r.getValue().getCacheControl());
+                assertEquals(contentDisposition, r.getValue().getContentDisposition());
+                assertEquals(contentEncoding, r.getValue().getContentEncoding());
+                assertEquals(contentLanguage, r.getValue().getContentLanguage());
+                assertEquals(finalContentType, r.getValue().getContentType());
```

Comment:
Instead of this, you can pass response into `validatePathProperties` like done in the sync version, so: 

`validatePathProperties(response, cacheControl, contentDisposition, contentEncoding, contentLanguage, null, contentType)).verifyComplete();`